odxtools/parameters/parameterbase.py

```diff
@@ -18,17 +18,29 @@ def __init__(self,
                  parameter_type,
                  long_name=None,
                  byte_position=None,
-                 bit_position=0,
+                 bit_position=None,
                  semantic=None,
                  description=None):
         self.short_name: str = short_name
         self.long_name: Optional[str] = long_name
         self.byte_position: Optional[int] = byte_position
-        self.bit_position: int = bit_position
+        self.bit_position: Optional[int] = bit_position
         self.parameter_type: str = parameter_type
         self.semantic: Optional[str] = semantic
         self.description: Optional[str] = description
 
+    @property
+    def byte_position_int(self) -> int:
+        if self.byte_position is None:
+            return 0
+        return self.byte_position
+
+    @property
+    def bit_position_int(self) -> int:
+        if self.bit_position is None:
+            return 0
+        return self.bit_position
+
```

Comment:
Those are extremely misleading, a byte/bit position None for parameter means "Encode/Decode this parameter just after the previous one"

Any logic that tries to imply 0 from a None will most likely be incorrect.

Reply:
the problem is that a lot of places in the code assumed that `bit_position` and 'byte_position' are integers. I introduced these functions to not having to do the checking manually all the time, but maybe it should be done that way...

Reply:
ok. I have removed them: https://github.com/mercedes-benz/odxtools/pull/86/commits/9fbc7388a6836ea99432edd3c189a9d8b26e8ad5. Hopefully everything still works as before...